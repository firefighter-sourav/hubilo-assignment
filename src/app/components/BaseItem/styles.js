import styled from 'styled-components'

import { colors } from 'app/themes/colorPalette'
import { fonts } from 'app/themes/fontPalette'

export const Wrapper = styled.div`
  padding: 2rem;
  box-shadow: 0 1px 3px rgba(0, 0, 0, 0.12), 0 1px 2px rgba(0, 0, 0, 0.24);
  transition: all 0.3s cubic-bezier(0.25, 0.8, 0.25, 1);
  margin-top: 10px;
  border-radius: 8px;
  &:hover {
    box-shadow: 0 4px 8px rgba(0, 0, 0, 0.16), 0 4px 4px rgba(0, 0, 0, 0.22);
  }
`
export const Title = styled.div`
  font-size: 18px;
  margin-bottom: 1rem;
  text-transform: capitalize;
  font-weight: bold;
  font-family: ${fonts.roboto};
`
export const Body = styled.div`
  font-size: 16px;
  color: ${colors.lightBlack};
  text-transform: capitalize;
`
