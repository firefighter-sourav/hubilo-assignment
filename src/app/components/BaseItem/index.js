import PropTypes from 'prop-types'
import React, { memo } from 'react'

import * as Styles from './styles'

/**
 * Generic item for all of similar types item view
 * Individual Item for list view
 * List can be optimized by react-virtualized/react-window
 */
const BaseItem = (props) => {
  const {
    style,
    item: { title, body },
  } = props

  return (
    <Styles.Wrapper style={style}>
      <Styles.Title>{title}</Styles.Title>
      {Object.keys(body).map((key, index) => (
        <Styles.Body key={body[key] + index}>
          {key}: {body[key]}
        </Styles.Body>
      ))}
    </Styles.Wrapper>
  )
}
BaseItem.propTypes = {
  item: PropTypes.object.isRequired,
  style: PropTypes.object,
}
BaseItem.defaultProps = {
  style: {},
}
export default memo(BaseItem)
