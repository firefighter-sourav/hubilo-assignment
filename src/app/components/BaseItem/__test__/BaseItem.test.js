import {
  cleanup,
  render as testRender,
  fireEvent,
} from '@testing-library/react'
import React from 'react'
import ReactDOM from 'react-dom'
import '@testing-library/jest-dom/extend-expect'
import renderer from 'react-test-renderer'

import BaseItem from '..'

afterEach(cleanup)

const data = {
  title: 'Hello',
  body: {
    id: 1,
    content: 'There',
  },
}

it('renders ConfigSelect without crasing', () => {
  const div = document.createElement('div')
  ReactDOM.render(<BaseItem item={data} />, div)
  ReactDOM.unmountComponentAtNode(div)
})

it('matches snapshot', () => {
  const tree = renderer.create(<BaseItem item={data} />).toJSON()
  expect(tree).toMatchSnapshot()
})
