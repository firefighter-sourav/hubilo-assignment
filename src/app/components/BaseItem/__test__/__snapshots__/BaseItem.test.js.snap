// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="sc-bdnxRM fkMDjp"
>
  <div
    className="sc-gtsrHT fSHCKA"
  >
    Hello
  </div>
  <div
    className="sc-dlnjwi cuPitm"
  >
    id
    : 
    1
  </div>
  <div
    className="sc-dlnjwi cuPitm"
  >
    content
    : 
    There
  </div>
</div>
`;
